mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java

```diff
@@ -174,6 +175,11 @@ public TaskExecutor(
             Executors.newFixedThreadPool(
                 Hardware.getNumberCPUCores(),
                 new ExecutorThreadFactory("taskexecutor-io"));
+
+        this.runtimeTaskExecutor =
```

Comment:
what benefits we get in separating this?

Reply:
Having main runtime task mixed in IO thread pool can potentially cause expected block/wait with other network IO ops?

Reply:
Hmm, have we seen any actual problems associated with this? I'm not convinced we need a dedicated thread to start the runtime task. I agree this is a problem if the io executor was single-threaded. But it looks like it's not.